### PR TITLE
feat(workflows): add release_move_ide.yml

### LINKED ISSUE #4288: [move-ide] Create a workflow to publish the Move VS Code extension
## Description

Create a workflow that can be triggered manually to publish the [move vs code extension](https://github.com/iotaledger/iota/tree/develop/external-crates/move/crates/move-analyzer/editors/code) to the visual studio marketplace using the token from https://github.com/iotaledger/iota/issues/4287
```
on:
  ## Allow triggering this workflow manually via GitHub CLI/web
  workflow_dispatch:
```
It should probably leverage this script, just with local builds instead of getting the binaries from the latest release to be sure that the TS code and binary is always in sync
https://github.com/iotaledger/iota/blob/develop/external-crates/move/crates/move-analyzer/editors/code/scripts/create.sh

https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh might help here with a build step to build the move analyzer binary before calling it

## Motivation

Have the extension published to the marketplace so users can install it from there

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add `release_move_ide.yml` with `create_from_local.sh` so we can publish the IOTA Move Visual Studio extension to the VS Marketplace.
It first builds the move analyzer binaries for macos (arm64+x86_64), ubuntu and windows and then creates the VS extension for each of them + one generic without a binary. For the generic one users have to build the move-analyzer binary themselves.
Also applied some minor patches from @valeriyr 

TODO: 
- [x] Update confluence workflows page when it's approved

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota/issues/4288

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running in the CI on this branch https://github.com/iotaledger/iota/actions/runs/12198732055?pr=4332
https://marketplace.visualstudio.com/items?itemName=iotaledger.iota-move :tada:

### CODE REVIEW

File: .github/workflows/release_move_ide.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      - name: Set os/arch variables
+        if: ${{ matrix.os == 'macos-latest' }}
+        shell: bash
+        run: |
+          export arch=$(uname -m)
+          export system_os=$(echo ${{ matrix.os }} | cut -d- -f1)
+          export os_type="${system_os}-${arch}"
+          echo "os_type=${system_os}-${arch}" >> $GITHUB_ENV
+
+      - name: Install nexttest (Windows)
```
Comment (thibault-martinez) [inline .github/workflows/release_move_ide.yml]: Why do we need nextest?

Comment (Thoralf-M) [reply .github/workflows/release_move_ide.yml]: Copy pasted too much :trollface:

File: .github/workflows/release_move_ide.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        if: ${{ matrix.os == 'windows-latest' }}
+        uses: taiki-e/install-action@375e0c7f08a66b8c2ba7e7eef31a6f91043a81b0 # pin@v2
+        with:
+          tool: nextest
+
+      - name: Cargo build for ${{ matrix.os }} platform
```
Comment (Thoralf-M) [inline .github/workflows/release_move_ide.yml]:
```suggestion
      - name: Cargo build for ${{ matrix.os }} platform
```

File: .github/workflows/release_move_ide.yml (hunk 1 above)
Comment (Thoralf-M) [reply .github/workflows/release_move_ide.yml]: Removed

File: external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
+#!/bin/zsh
+# Copyright (c) Mysten Labs, Inc.
+# Modifications Copyright (c) 2024 IOTA Stiftung
```
Comment (DaughterOfMars) [inline external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh:3]: Is this actually a mysten script that we are pulling?

Comment (Thoralf-M) [reply external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh:3]: Yeah, but modified https://github.com/MystenLabs/sui/blob/5c06cf64cddbe537bcc992231d94b9211b067ece/external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh#L4
They seem to run this locally manually and not in the CI

File: external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+# This script is meant to be executed on MacOS (hence zsh use - to get associative arrays otherwise
```
Comment (lzpap) [inline external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh]: It's being run on self-hosted (ubuntu), is this comment still relevant?

File: external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+# unavailable in the bundled bash version).
```
Comment (Thoralf-M) [inline external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh]:
```suggestion
```

File: external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh (hunk 4 above)
Comment (Thoralf-M) [reply external-crates/move/crates/move-analyzer/editors/code/scripts/create_from_local.sh]: No, removed in https://github.com/iotaledger/iota/pull/4332/commits/3571444b235d546ce5430acf1d01364358147af1

**Review by miker83z — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]
We should edit the README. It's not the best imo

**Comment by thibault-martinez** [comment]
> We should edit the README. It's not the best imo

Which README?

**Comment by DaughterOfMars** [comment]
> > We should edit the README. It's not the best imo
> 
> Which README?

Whichever one is in the published page

**Pull request merged**